bot/resources/tags/type-hint.md

```diff
@@ -0,0 +1,16 @@
+**Type Hints**
+
+A typehint indicates what type something should be. For example,
+```python
+def add(a: int, b: int) -> int:
+    sum: int = a + b
+    return sum
+```
+In this case, `a` and `b` are expected to be ints, and the function returns an int. We also declare an intermediate variable `sum`, which we indicate to be an int.
```

Comment:
Should probably mention that the `sum` variable can be inferred (or guessed could be a easier word to understand) by whatever static type checker the user is using.